src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -550,8 +458,9 @@ public RuntimeExecutionResults Process()
             Debug.Assert(extensionDirectory != null);
 
             var assemblyLoader = createInfo.AssemblyLoader ?? new AssemblyLoader();
+            var versionChecker = createInfo.Versioning ?? VersionChecker.Create();
 
-            var assemblyDiscovery = new AssemblyExtensionDiscovery(assemblyLoader);
+            var assemblyDiscovery = new AssemblyExtensionDiscovery(assemblyLoader, versionChecker);
```

Comment:
>, versionChecker [](start = 81, length = 16)

since VersionChecker is a concrete type with no parameters, should we have an AssemblyExtensionDiscovery ctor that doesn't require this to be passed in - just creates it instead?